### ISSUE TITLE
Crash when running in a directory containing 8 files

### ISSUE THREAD

**Issue by gadanidis** [nimaipatel/brn#5]
When running `brn` in this git repository after running `make`, it crashes with the following error:

```
malloc(): invalid next size (unsorted)
[1]    85812 IOT instruction (core dumped)  brn
```

The directory looks like this:
```
.git
.gitignore
brn
brn.c
brn-example.gif
LICENSE
Makefile
README.md
```

I have not been able to reproduce the error in any other git repositories. The error disappears when any file is removed, or if another file is added to the repository. It does not occur on a fresh clone (with no `brn` executable) and also does not occur in other directories containing the `brn` executable. I know nothing about C so I'm not sure what could be causing this or how to debug further.

This is obviously not really affecting my workflow, I just noticed it because after installing I tried to test it out by renaming the readme, and got this error instead.

Thanks for brn, I'm finding it very useful!

**Comment by nimaipatel** [comment]
That's very strange. I'm noticing it on a fresh clone too but not in my original development repository. Thanks for reporting this issue, I'll dig into it.

**Comment by gadanidis** [comment]
On further use of brn, I've noticed that this seems to happen in any directory with 8 files/subdirectories, so that's why it only happens in the brn repo after running `make`, because the resulting directory contains 8 files. I can reliably reproduce with:

```
mkdir test && cd test
touch 1 2 3 4 5 6 7 8
brn
```

and

```
mkdir test && cd test
mkdir 1 2 3 4 5 6 7 8
brn
```

**Comment by nimaipatel** [comment]
The issue seems to arise because of commit ccda59ed980c66a9e950fb4687fef681cb538d6e.
I'll revert it for now and fix the 100 file limit in a while. Once again, thanks a lot for testing and reporting!

**Comment by nimaipatel** [comment]
Fixed in 82ae79d8e5b201b0fe4e8df4286a452c42ba968c and 454e2130c5621745309d47c3c1c82bf3226fe61f.
